File: bench/sieve.lua
-- BOT-TRANSLATED (with tweaks).

local function reset(flags, len)
  for ind = 1, len do
    flags[ind] = 1
  end
end

local function find_prime(flags, len)
  reset(flags, len)

  local num = 0
  local step = 3

  for ind = 1, len do
    if flags[ind] ~= 0 then
      for ind1 = ind + step, len, step do
        flags[ind1] = 0
      end

      num = num + 1
    end

    step = step + 2
  end

  return num
end

local function run(len, runs)
  local flags = {}
  local out = 0
  for _ = 1, runs do
    out = find_prime(flags, len)
  end
  if out ~= 1899 then error("mismatch: expected 1899; got " .. out) end
end

run(8192, 16384)

File: bench/sieve.mjs
// BOT-TRANSLATED (with tweaks).

function reset(flags) {flags.fill(1)}

function findPrime(flags) {
  reset(flags)

  let num = 0
  let step = 3

  for (let ind = 0; ind < flags.length; ind++) {
    if (flags[ind]) {
      for (let ind1 = ind + step; ind1 < flags.length; ind1 += step) {
        flags[ind1] = 0
      }
      num++
    }
    step += 2
  }

  return num
}

function run(flags, runs) {
  let out = 0
  for (let ind = 0; ind < runs; ind++) out = findPrime(flags)
  if (out !== 1899) throw Error(`mismatch: expected 1899; got ${out}`)
}

run(new Uint8Array(8192), 16384)
